File: Demo/NUITests/UIKitTests/UIActivityIndicatorView/NUIActivityIndicatorRendererTests.m
//
//  NUIActivityIndicatorRendererTests.m
//  NUIDemo
//
//  Created by Gomez, Antonio on 25/04/2016.
//  Copyright © 2016 Tom Benner. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "NUIConstants.h"
#import "NUIFileMonitor.h"
#import "UIView+NUI.h"
#import "NUIDemo-Swift.h"


@interface NUIActivityIndicatorRendererTests : XCTestCase
@property (strong, nonatomic) UIActivityIndicatorView *sut;
@end

@implementation NUIActivityIndicatorRendererTests

- (void)setUp {
    [super setUp];
    
    [NUISettings initWithStylesheet:@"TestTheme.NUI"];
    
    _sut = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
    [_sut applyNUI];
    
}

- (void)tearDown {
    _sut = nil;
    [super tearDown];
}

// color (Color)
- (void)testSetColor {
    XCTAssertEqualObjects(_sut.color, [UIColor redColor], @"NUI should set color activity indicator color");
}

@end

File: Demo/NUITests/UIKitTests/UIActivityIndicatorView/NUIActivityIndicatorRendererTests.swift
//
//  NUIActivityIndicatorRendererTests.swift
//  NUIDemo
//
//  Created by Sergej Jaskiewicz on 08/04/16.
//  Copyright © 2016 Tom Benner. All rights reserved.
//

import XCTest
@testable import NUIDemo

class NUIActivityIndicatorRendererTests: XCTestCase {
    
    var sut: UIActivityIndicatorView!
    
    override func setUp() {
        super.setUp()
        
        NUISettings.initWithStylesheet("TestTheme.NUI")
        sut = UIActivityIndicatorView(activityIndicatorStyle: .gray)
        sut.applyNUI()
    }

    func testSetColor() {
        XCTAssertEqual(sut.color, .red, "NUI should set color activity indicator color")
    }
}
